### PR TITLE
AtlasMap native support #1989

### PR DESCRIPTION
**Pull request by zbendhiba**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/extensions/atlasmap.adoc [hunk 1]
```diff
@@ -31,3 +31,27 @@ Please refer to the above link for usage and configuration details.
...
+* XML
+* Json
+* CSV
+
+The _ADM archive file_ or _mapping definition JSON file_ used in the _AtlasMap route_ should be located in the _resources_ repository.
+In native mode, include this file for native build. Example:
+[source,properties]
+----
+quarkus.native.additional-build-args = -H:IncludeResources=atlas-mapping.json
+----
```
Comment (aldettinger) [inline docs/modules/ROOT/pages/reference/extensions/atlasmap.adoc:54]: Maybe one could use https://quarkus.io/guides/building-native-image#quarkus-native-pkg-native-config_quarkus.native.resources.includes instead.

File: integration-tests/atlasmap/src/test/java/org/apache/camel/quarkus/component/atlasmap/it/AtlasmapTest.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    @Test
+    void testJava2JsonWithJson() {
+        Person person = new Person("foo", "bar", 35);
+        given()
+                .contentType(ContentType.JSON)
+                .body(person)
+                .when()
+                .get("/json/java2json")
+                .then()
+                .body("name2", equalTo("bar"))
```
Comment (aldettinger) [inline integration-tests/atlasmap/src/test/java/org/apache/camel/quarkus/component/atlasmap/it/AtlasmapTest.java:42]: Any reason we don't assert name1 ?

**Review by aldettinger — APPROVED** [review]
Looks good to me, I may have spotted minor things. That's quite a PR, congrats @zbendhiba .

**Review by jamesnetherton — APPROVED** [review]
Nice work :sunglasses:

File: extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        items.add(new ReflectiveClassBuildItem(true, true, XmlField.class));
+        items.add(new ReflectiveClassBuildItem(true, true, XmlNamespace.class));
+        items.add(new ReflectiveClassBuildItem(true, true, XmlNamespaces.class));
+        items.add(new ReflectiveClassBuildItem(true, true, CsvDataSource.class));
+        items.add(new ReflectiveClassBuildItem(true, true, CsvField.class));
+        items.add(new ReflectiveClassBuildItem(true, true, ADMDigest.class));
+        items.add(new ReflectiveClassBuildItem(true, true, DataSourceMetadata.class));
+        items.add(new ReflectiveClassBuildItem(false, true, true, InspectionType.class));
+        items.add(new ReflectiveClassBuildItem(true, true, ValueContainer.class));
+        return items;
```
Comment (lburgazzoli) [inline extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java:356]: Maybe as follow up we may use jandex to discover the types we need to register for reflection, best would be if atlasmap weere providing the jamdex index but if not, we can explicitly index atlasmap jars

**Review by lburgazzoli — APPROVED** [review]

**Review by ppalaga — APPROVED** [review]
Nice amount of work, thanks!

File: integration-tests/atlasmap/src/test/java/org/apache/camel/quarkus/component/atlasmap/it/AtlasmapTest.java (hunk 2 above)
Comment (zbendhiba) [reply integration-tests/atlasmap/src/test/java/org/apache/camel/quarkus/component/atlasmap/it/AtlasmapTest.java:42]: just forgot it

File: docs/modules/ROOT/pages/reference/extensions/atlasmap.adoc (hunk 1 above)
Comment (zbendhiba) [reply docs/modules/ROOT/pages/reference/extensions/atlasmap.adoc:54]: thanks, I didn't know this one !!

File: extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java (hunk 3 above)
Comment (zbendhiba) [reply extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java:356]: @lburgazzoli : I've created an issue to follow up  with this one : [2241](https://github.com/apache/camel-quarkus/issues/2241)

**Comment by zbendhiba** [comment]
> Looks good to me, I may have spotted minor things. That's quite a PR, congrats @zbendhiba .
I've just commit the changes. Thanks @aldettinger  for the review comments

**Pull request merged**